iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/CheckUtils.java

```diff
@@ -21,23 +21,32 @@
 
 import java.util.Collection;
 import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
 import java.util.function.BiConsumer;
+import java.util.stream.Stream;
 import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.tree.impl.TextRanges;
 import org.sonar.iac.docker.symbols.ArgumentResolution;
 import org.sonar.iac.docker.tree.api.CommandInstruction;
 import org.sonar.iac.docker.tree.api.DockerTree;
 import org.sonar.iac.docker.tree.api.Flag;
 import org.sonar.iac.docker.tree.api.HasArguments;
+import org.sonar.iac.docker.tree.api.ShellForm;
+import org.sonar.iac.docker.tree.api.Variable;
 
 public final class CheckUtils {
+  private static final ResolvedCommandPartParser RESOLVED_COMMAND_PART_PARSER = ResolvedCommandPartParser.create();
```

Comment:
It may seem an overkill, but I couldn't think of an easy way to parse the command without duplicating the logic for splitting shell form and creating artificial `ArgumentResolution`s to map text ranges